src/plugins/intel_gpu/src/graph/gather_tree.cpp

```diff
@@ -23,6 +24,32 @@ layout gather_tree_inst::calc_output_layout(gather_tree_node const& node, kernel
     return input_layout;
 }
 
+template<typename ShapeType>
+std::vector<layout> gather_tree_inst::calc_output_layouts(gather_tree_node const& /*node*/, const kernel_impl_params& impl_param) {
+    auto desc = impl_param.typed_desc<gather_tree>();
+    auto input0_layout = impl_param.get_input_layout(0);
+
+    auto output_type = input0_layout.data_type;
+    if (impl_param.has_fused_primitives()) {
+        output_type = impl_param.get_fused_output_layout().data_type;
+    }
+
+    ov::op::v1::GatherTree op;
+
+    std::vector<ShapeType> output_shapes = {ShapeType()};
+    std::vector<ShapeType> input_shapes = {
+        impl_param.get_input_layout(0).get<ShapeType>(),
+        impl_param.get_input_layout(1).get<ShapeType>(),
+        impl_param.get_input_layout(2).get<ShapeType>(),
+        impl_param.get_input_layout(3).get<ShapeType>(),
+    };
+    ov::op::v1::shape_infer(&op, input_shapes, output_shapes);
+
+    format output_format = format::adjust_to_rank(input0_layout.format, output_shapes[0].size());
```

Comment:
output_shapes[0].size() is only working when rank is static. In your test case, it is working well because rank is static. but if rank is dynamic, it will be failed. could you please check it again?

If you add this test params { `layout{ov::PartialShape::dynamic(), data_types::f32, format::bfyx},
            layout{ov::PartialShape::dynamic(), data_types::f32, format::bfyx},
            layout{ov::PartialShape::dynamic(1), data_types::f32, format::bfyx},
            layout{ov::PartialShape::dynamic(0), data_types::f32, format::bfyx},
            layout{ov::PartialShape::dynamic(), data_types::f32, format::bfyx}` }, you can find error related rank

Reply:
I added some dynamic cases but this was fine.

Reply:
Your test params have rank but i means that create ov::ParitalShape::Dynamic() without rank.

Reply:
We don't target dynamic rank support at the moment, so `output_shapes[0].size()` is fine